### ISSUE TITLE
connect ENOBUFS - too many concurrent TCP connections

### ISSUE THREAD

**Issue by cybermelons** [tsaibermelon/gg-struggle#18]
But then there's a part which throws another error:

```
[2021-07-26T20:50:46.964] [INFO] default - [DB] Dumping request to \AppData\Local\Temp/gg-struggle/dumps/b627588615559e35366585b10f1b5e2e203a4a7f.gameReq.dump

events.js:292
      throw er; // Unhandled 'error' event
      ^

Error: connect ENOBUFS 127.0.0.1:443 - Local (undefined:undefined)
    at internalConnect (net.js:921:16)
    at defaultTriggerAsyncIdScope (internal/async_hooks.js:430:12)
    at net.js:1009:9
    at processTicksAndRejections (internal/process/task_queues.js:75:11)
Emitted 'error' event on ClientRequest instance at:
    at TLSSocket.socketErrorListener (_http_client.js:469:9)
    at TLSSocket.emit (events.js:315:20)
    at emitErrorNT (internal/streams/destroy.js:106:8)
    at emitErrorCloseNT (internal/streams/destroy.js:74:3)
    at processTicksAndRejections (internal/process/task_queues.js:80:21) {
  errno: -4060,
  code: 'ENOBUFS',
  syscall: 'connect',
  address: '127.0.0.1',
  port: 443
}
```

Prior to that, my log is spammed with this:

```
[2021-07-26T20:46:07.656] [INFO] default - [PROXY] /api/sys/get_env POST b627588615559e35366585b10f1b5e2e203a4a7f
[2021-07-26T20:46:07.657] [INFO] default - [DB] Dumping request to \AppData\Local\Temp/gg-struggle/dumps/b627588615559e35366585b10f1b5e2e203a4a7f.gameReq.dump
[2021-07-26T20:46:07.657] [INFO] default - [CACHE] Miss: /api/sys/get_env POST b627588615559e35366585b10f1b5e2e203a4a7f
[2021-07-26T20:46:07.674] [INFO] default - [PROXY] /api/sys/get_env POST b627588615559e35366585b10f1b5e2e203a4a7f
[2021-07-26T20:46:07.674] [INFO] default - [DB] Dumping request to \AppData\Local\Temp/gg-struggle/dumps/b627588615559e35366585b10f1b5e2e203a4a7f.gameReq.dump
[2021-07-26T20:46:07.674] [INFO] default - [CACHE] Miss: /api/sys/get_env POST b627588615559e35366585b10f1b5e2e203a4a7f
[2021-07-26T20:46:07.680] [INFO] default - [PROXY] /api/sys/get_env POST b627588615559e35366585b10f1b5e2e203a4a7f
[2021-07-26T20:46:07.680] [INFO] default - [DB] Dumping request to \AppData\Local\Temp/gg-struggle/dumps/b627588615559e35366585b10f1b5e2e203a4a7f.gameReq.dump
[2021-07-26T20:46:07.681] [INFO] default - [CACHE] Miss: /api/sys/get_env POST b627588615559e35366585b10f1b5e2e203a4a7f
```

It also additionally spams this one around 30 seconds after:

```
[2021-07-26T21:02:08.830] [INFO] default - [PROXY] /api/sys/get_news POST 722ed30712c19b713ff3a993598e1eb9d95fce65
[2021-07-26T21:02:08.830] [INFO] default - [DB] Dumping request to \AppData\Local\Temp/gg-struggle/dumps/722ed30712c19b713ff3a993598e1eb9d95fce65.gameReq.dump
[2021-07-26T21:02:08.830] [INFO] default - [CACHE] Miss: /api/sys/get_news POST 722ed30712c19b713ff3a993598e1eb9d95fce65
```

_Originally posted by @priprii in https://github.com/tsaibermelon/gg-struggle/issues/17#issuecomment-886978497_

**Comment by cybermelons** [comment]
I'm guessing it's to do with https://github.com/nodejs/node-v0.x-archive/issues/3269  or https://stackoverflow.com/questions/10603860/very-simple-node-js-client-throws-error-enobufs-after-many-http-requests

Possibly too many connections open from using `https.request`. Maybe we should disable by `agent:false` 

```
var options = {agent:false, host:"ggst-game.guiltygear.com", port:443, /*...*/ };
```

**Comment by cybermelons** [comment]
Change https://github.com/tsaibermelon/gg-struggle/blob/main/server/gg-struggle.js#L91 : 

```  fetchGg = (gameReq, callback) => {
    const key = gameReq.key
    const options = {
      agent: false,
      hostname: this.ggHost,
      port: 443,
      path: gameReq.url,
      method: gameReq.method,
```

**Comment by cybermelons** [comment]
I suspect the v2.0 errors are due to node 16. I packaged v1.0 with the node module `pkg` and used `node14-win` as the build arch.

**Comment by priprii** [comment]
> I suspect the v2.0 errors are due to node 16. I packaged v1.0 with the node module `pkg` and used `node14-win` as the build arch.

My node version is 14.16.1

**Comment by cybermelons** [comment]
> > I suspect the v2.0 errors are due to node 16. I packaged v1.0 with the node module `pkg` and used `node14-win` as the build arch.
> 
> My node version is 14.16.1

welp, there goes that idea.

**Comment by cybermelons** [comment]
oh I know what's happening. the dns resolution isn't working and it's looping back to itself. specifically, shouldn't be using `dns.setServers()` but something else.

From the [DNS node doc](https://nodejs.org/api/dns.html#dns_dns_setservers_servers): 
> The dns.setServers() method affects only dns.resolve(), dns.resolve*() and dns.reverse() (and specifically not dns.lookup()).

**Comment by cybermelons** [comment]
yup, caused by loopback. Fixed in #20
